lib/cylc/config.py

```diff
@@ -1450,9 +1443,28 @@ def generate_edges( self, lexpression, left_nodes, right, seq, suicide=False ):
             conditional = True
 
         for left in left_nodes:
-            if left is not None:
-                e = graphing.edge( left, right, seq, suicide, conditional )
-                self.edges.append(e)
+            if left is None:
+                continue
+            if right is not None:
+                # Check for self-suicide and self-edges.
+                if left == right or left.startswith(right + ':'):
+                    # (This passes inter-cycle offsets: left[-P1D] => left)
+                    # (TODO - but not explicit null offsets like [-P0D]!)
```

Comment:
If someone writes ```foo[-P0D] => foo``` it will not be detected as a self-edge, but will result in a more obtuse error later (due to attempting to style an edge that has been removed by ```strict=True```).  Do you think we need to handle this?  If so, is it possible to determine if different null intervals are equal (e.g. P0D == P0Y)?